sinon/test/TestSinonSpy.py

```diff
@@ -141,34 +143,57 @@ def test051_calledBefore_calledAfter_normal(self):
         os.system("cd")
         spy2()
         sinon.g.B_func()
+        self.assertFalse(spy1.calledBefore(spy1))
+        self.assertFalse(spy2.calledBefore(spy2))
+        self.assertFalse(spy3.calledBefore(spy3))
+        self.assertFalse(spy1.calledAfter(spy1))
+        self.assertFalse(spy2.calledAfter(spy2))
+        self.assertFalse(spy3.calledAfter(spy3))
         self.assertTrue(spy1.calledBefore(spy2))
         self.assertTrue(spy1.calledBefore(spy3))
         self.assertTrue(spy2.calledBefore(spy3))
+        self.assertFalse(spy1.calledAfter(spy2))
+        self.assertFalse(spy1.calledAfter(spy3))
+        self.assertFalse(spy2.calledAfter(spy3))
         self.assertTrue(spy2.calledAfter(spy1))
         self.assertTrue(spy3.calledAfter(spy1))
         self.assertTrue(spy3.calledAfter(spy2))
+        self.assertFalse(spy2.calledBefore(spy1))
+        self.assertFalse(spy3.calledBefore(spy1))
+        self.assertFalse(spy3.calledBefore(spy2))
 
     @sinontest
     def test052_calledBefore_nothing_called(self):
         spy1 = SinonSpy(os, "system")
         spy2 = SinonSpy()
         spy3 = SinonSpy(B_func)
-        with self.assertRaises(Exception) as context:
-            self.assertFalse(spy1.calledBefore(spy2))
-        with self.assertRaises(Exception) as context:
-            self.assertFalse(spy2.calledAfter(spy1))
+        self.assertFalse(spy1.calledBefore(spy2))
+        self.assertFalse(spy2.calledBefore(spy1))
+        self.assertFalse(spy2.calledAfter(spy1))
+        self.assertFalse(spy1.calledAfter(spy2))
  
     @sinontest
     def test053_calledBefore_calledAfter_recalled_method(self):
         spy1 = SinonSpy(os, "system")
-        spy2 = SinonSpy()
-        os.system("cd")
-        spy2()
-        os.system("cd")
-        self.assertTrue(spy1.calledBefore(spy2))
-        self.assertTrue(spy1.calledAfter(spy2))
-        self.assertTrue(spy2.calledBefore(spy1))
-        self.assertTrue(spy2.calledAfter(spy1))
+        try:
```

Comment:
I am not sure why you used try/finally here, can you explain your intention?

Reply:
Hi Kir,

The test was failing while I was working on the `CALLQUEUE` refactoring, and the test failure was leaking into other tests and causing them to fail too. That's why I added the try/catch. I can remove the try/catch, but based on what you're saying there seems to be a bug in the `@sinontest` handling, since it doesn't seem to always work. 

--Jonathan